src/main/java/net/sigmarik/abilitymod/mixin/PlayerEntityMixin.java

```diff
@@ -29,61 +30,107 @@
 @Mixin(PlayerEntity.class)
 public abstract class PlayerEntityMixin extends LivingEntity {
 
+    @Shadow public abstract PlayerInventory getInventory();
+
+    @Shadow public abstract Iterable<ItemStack> getArmorItems();
+
     protected PlayerEntityMixin(EntityType<? extends LivingEntity> entityType, World world) {
         super(entityType, world);
     }
 
     private void tickFearOfWater() {
-        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_FEAR_OF_WATER) && isWet()) {
-            damage(getDamageSources().magic(), 3);
+        if (!ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_FEAR_OF_WATER) && isWet()) {
+            return;
         }
+        damage(getDamageSources().magic(), 3);
     }
 
     private void tickAddiction() {
-        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_ADDICTION)) {
-            ServerState.tickAddictionTimer((PlayerEntity)(Object)this);
+        if (!ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_ADDICTION)) {
+            return;
+        }
+        ServerState.tickAddictionTimer((PlayerEntity)(Object)this);
 
-            int timer = ServerState.getAddictionTimer((PlayerEntity)(Object)this);
+        int timer = ServerState.getAddictionTimer((PlayerEntity)(Object)this);
 
-            if (timer < AbilityMod.ADDICTION_WARNING_TIMER) {
-                addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, timer));
-            }
-            if (timer == 0) {
-                addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.HUNGER, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.NAUSEA, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.DARKNESS, 20));
-                addStatusEffect(new StatusEffectInstance(StatusEffects.SPEED, 20));
-            }
+        if (timer < AbilityMod.ADDICTION_WARNING_TIMER) {
+            addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, timer));
+        }
+        if (timer == 0) {
+            addStatusEffect(new StatusEffectInstance(StatusEffects.WEAKNESS, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.HUNGER, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.NAUSEA, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.DARKNESS, 20));
+            addStatusEffect(new StatusEffectInstance(StatusEffects.SPEED, 20));
         }
     }
 
     private void tickBoatMagnet() {
-        if (ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_BOAT_MAGNET) && hasVehicle() &&
+        if (!ServerState.hasTrait((PlayerEntity)(Object)this, AbilityMod.TRAIT_BOAT_MAGNET) && hasVehicle() &&
```

Comment:
You did not inverse the condition. The code above (and a few lines below) do not represent the same logic there was before.
This code just does not work correctly.